server/routes/getExposureOverTime.ts

```diff
@@ -0,0 +1,146 @@
+import Sequelize from 'sequelize';
+import { Exposure, BlockNumber, AccountId, IndividualExposure } from '@polkadot/types/interfaces';
+import { Request, Response, NextFunction } from 'express';
+import { Errors } from './getOffences';
+const Op = Sequelize.Op;
+
+interface IEventData {
+    stash: AccountId;
+    exposure: Exposure;
+    block_number: BlockNumber;
+}
+
+let stakeOverTime;
+// eslint-disable-next-line prefer-const
+let validators: { [key: string]: any[] } = {};
+
+// COMMON
+const getStakeOverTime = async (models, req: Request, res: Response, next: NextFunction) => {
+  const { chain, stash } = req.query;
+  let { startDate, endDate } = req.query;
+
+  if (!chain) return next(new Error(Errors.ChainIdNotFound));
+  const chainInfo = await models.Chain.findOne({ where: { id: chain } });
+  if (!chainInfo) return next(new Error(Errors.InvalidChain));
+
+  // if start and end date isn't given, we set it for a year from now
+  if (typeof startDate === 'undefined' || typeof endDate === 'undefined') {
+    startDate = new Date();
+    startDate.setFullYear(startDate.getFullYear() - 1);
+    startDate = new Date(startDate);
+    endDate = new Date();
+  }
+
+  const where: any = {
+    '$ChainEventType.chain$': chain,
+  };
+
+  // if stash is given
+  // eslint-disable-next-line no-unused-expressions
+  stash && (where['$HistoricalValidatorStatistic.stash$'] = stash);
```

Comment:
please avoid this syntax for backend code, you should just do `if (stash) where['$...'] = stash`

Reply:
Thanks for letting me know; I'll take care next time.